subprojects/distributions-full/build.gradle.kts

```diff
@@ -30,3 +51,19 @@ tasks.register<Copy>("copyDistributionsToRootBuild") {
     from(layout.buildDirectory.dir("distributions"))
     into(rootProject.layout.buildDirectory.dir("distributions"))
 }
+
+tasks.register<Jar>("jarPublicApi") {
+    from(configurations.runtimeClasspath.get().incoming.artifactView {
+        attributes {
+            attribute(filteredAttribute, Filtering.PUBLIC_API)
+        }
+        componentFilter {
+            it is ProjectComponentIdentifier &&
+                // FIXME Why is this dependency present here? Can we exclude it better?
+                !it.buildTreePath.equals(":build-logic:kotlin-dsl-shared-runtime")
+        }
+    }.files)
+    destinationDirectory = layout.buildDirectory.dir("public-api-jar")
+    // FIXME This is required because package-info.class files are duplicated
+    duplicatesStrategy = DuplicatesStrategy.WARN
```

Comment:
I imagine this will be pretty loud and annoying

Reply:
Actually, we don't need this. It appears that we only have duplicate `package-info.class` files in implementation packages, so all good. But even there we have like ten.